xarray/util/generate_reductions.py

```diff
@@ -3,294 +3,398 @@
-
-    kwarg = collections.namedtuple("kwarg", "docs kwarg call")
-    if skipna:
-        skip_na = kwarg(
-            docs=textwrap.indent(_SKIPNA_DOCSTRING, "        "),
-            kwarg="\n        skipna: bool = True,",
-            call="\n            skipna=skipna,",
-        )
-    else:
-        skip_na = kwarg(docs="", kwarg="", call="")
-
-    if method in MIN_COUNT_METHODS:
-        min_count = kwarg(
-            docs=textwrap.indent(_MINCOUNT_DOCSTRING, "        "),
-            kwarg="\n        min_count: Optional[int] = None,",
-            call="\n            min_count=min_count,",
-        )
-    else:
-        min_count = kwarg(docs="", kwarg="", call="")
-
-    return TEMPLATE_REDUCTION.format(
-        obj=obj,
-        docref=docref,
-        method=method,
-        array_method=array_method,
-        extra_dim="""If ``None``, will reduce over all dimensions
-            present in the grouped variable.""",
-        extra_args="",
-        skip_na=skip_na,
-        min_count=min_count,
-        numeric_only_call=numeric_only_call,
-        example=example_generator(obj=obj, method=method),
-    )
-
-
-def render(obj: str, cls: str, docref: str, example_generator: Callable):
-    yield CLASS_PREAMBLE.format(obj=obj, cls=cls)
-    yield generate_method(
-        obj,
-        method="count",
-        docref=docref,
-        skipna=False,
-        example_generator=example_generator,
-    )
-    for method in BOOL_REDUCE_METHODS:
-        yield generate_method(
-            obj,
-            method=method,
-            docref=docref,
-            skipna=False,
-            array_method=f"array_{method}",
-            example_generator=example_generator,
-        )
-    for method in NAN_REDUCE_METHODS:
-        yield generate_method(
-            obj,
-            method=method,
-            docref=docref,
-            skipna=True,
-            example_generator=example_generator,
-        )
+            extra_kwargs = ""
+        return f"""        return self.reduce(
+            duck_array_ops.{method.array_method},
+            dim=dim,{extra_kwargs}
+            keep_attrs=keep_attrs,
+            **kwargs,
+        )"""
+
+
+REDUCTION_METHODS = (
+    Method("count"),
+    Method("all", bool_reduce=True),
+    Method("any", bool_reduce=True),
+    Method("max", extra_kwargs=(skipna,)),
+    Method("min", extra_kwargs=(skipna,)),
+    Method("mean", extra_kwargs=(skipna,), numeric_only=True),
+    Method("prod", extra_kwargs=(skipna, min_count), numeric_only=True),
+    Method("sum", extra_kwargs=(skipna, min_count), numeric_only=True),
+    Method("std", extra_kwargs=(skipna, ddof), numeric_only=True),
+    Method("var", extra_kwargs=(skipna, ddof), numeric_only=True),
+    Method("median", extra_kwargs=(skipna,), numeric_only=True),
+)
+
+
+@dataclass
+class DataStructure:
+    name: str
+    docstring_create: str
+    example_var_name: str
+    numeric_only: bool = False
+
+
+DatasetObject = DataStructure(
```

Comment:
nit: this an global variable, so the naming convection is `ALL_CAPS`:
```suggestion
DATASET_OBJECT = DataStructure(
```